### ISSUE TITLE
Accessibility Issues

### ISSUE THREAD

**Issue by thornbill** [jellyfin/jellyfin-web#442]
**Describe the bug**

Tracking some accessibility issues reported by a user on [Reddit](https://old.reddit.com/r/jellyfin/comments/cuk9fe/accessibility_in_jellyfin/).

About the web UI, I would like to make a few suggestions:

1. In all menu items, The screen reading software seems to interpret the name of the icon assigned to the menu, so for example a navigation in the main menu reads like this (my jellyfin is in Spanish): "home Inicio Medios video_library Movies Administrador dashboard Panel de control mode_edit Metadatos" Ideally it should display just the names of the links, but not the text used for the icons.

2. There are some cases where the screen reader detects extra buttons without labels. I am not sure whether this is a problem in the page DOM or if all of those buttons are really unlabeled. An example of this is when you display a movie information, after the section which displays the credits, it shows 3 unlabeled buttons after the author name. I noticed If I press the first button, it shows basic information about the author, if I press the second or third button, it displays options such as update metadata, edit images, add to collection, and so on. If you want I can take a deeper look to the code in the DOM the next week when I will have some free time so I can explain it a bit better.

3. When you play a media file, the player does not appear to the screen reader's eyes. It would be nice if the player would be able to be managed by the application, about this I have no idea exactly why it happens, I guess it may be something related to the player being in a DIV and some CSS related things, but honestly I have no clue as still need to check the page.

Finally, I thought unlabeled buttons may be interpreted in such way because they may be just icons, without text, in such case i think the ARIA specification (Accessible Rich Internet Applications) has some info about how to make the screen reader to read text that is not visible to users, such information may be useful in case those buttons are just icons, as the screen reader will be able to interpret them anyway.

**Comment by manuelcortez** [comment]
Hey there,
I am that reddit user who posted about accessibility in Jellyfin. I think this can be improved by using a few recommendations. I have no idea about node.js and stuff, so I put these suggestions here. Most of those should be relatively easy to implement.

The first item on the list I have originally wrote about is already fixed. Now, the screen reading software actually reads properly all menu items, without adding the icon name in the menu, so this should be skipped as it works as expected. Now, there are my couple suggestions that would make the web interface a lot more accessible to screen reader users:

1. I understand there are some unlabeled buttons (I mean, buttons without texts) around the web interface. For example, I have found those by digging in the DOM:  
	<button type="button" is="paper-icon-button-light" data-ripple="false" data-direction="left" class="emby-scrollbuttons-button paper-icon-button-light hide" disabled=""><span class="material-icons chevron_left"></span></button>  
	<button type="button" is="paper-icon-button-light" data-ripple="false" data-direction="right" class="emby-scrollbuttons-button paper-icon-button-light hide" disabled=""><span class="material-icons chevron_right"></span></button>  
I guess those are something related to going back and forward in the navigation (?) by the "left" and "right" in the code, but certainly I am not sure. My software just reads "button". If you could add some text in the Span element or something, that would make it much better. If this is not possible due to the control looking weird, there are certain ways to add aria information which will be available to assistive technologies only.
2. It also would be super nice if you could use sections in the code (by using <main> in the main section of any page, or <section> in headers). Most assistive technologies allow us to jump between sections and the <main> tag easier than navigating with the arrow keys, making the app read all the main menu again, until we found the useful stuff. By using sections, that would not modify the visual appearance of the Web UI, while will make for us much easier to navigate in the web UI.
3. This point should be considered as a bonus, but if you render external (srt for example) subtitles in a DIV container or alike, you might consider adding role="alert" to the element rendering subtitles, so screen readers will read automatically every new change rendered right there. I have suggested [the same for plex web](https://forums.plex.tv/t/plex-web-read-subtitles-automatically-for-blind-people-only-one-change-is-needed/579266) with good results after implementation.

I think after adding those little suggestions, the web UI would be much more accessible than what it is so far. Right now it is pretty usable, but it might be better for users of any assistive technology.
